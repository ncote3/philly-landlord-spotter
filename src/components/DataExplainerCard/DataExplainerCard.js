import React from "react";
import Card from "react-bootstrap/Card";
import "./DataExplainerCard.scss";

export default function DataExplainerCard(props) {
  const { title, example, description, dataReference } = props;
  return (
    <Card border={"dark"}>
      <Card.Header className="dataDescCardBold dataDescCardHeader">
        {title}
      </Card.Header>
      <Card.Body>
        <Card.Text>
          <p>
            <span className={"dataDescCardBold"}>Example: </span>
            <span className={"dataDescCardItalic"}>{example}</span>
          </p>
          <p>
            <span className={"dataDescCardBold"}>Description:</span>
            {description}
          </p>
          <p>
            <span className={"dataDescCardBold"}>Data Reference: </span>
            <span className={"dataDescCardItalic"}>{dataReference}</span>
          </p>
        </Card.Text>
      </Card.Body>
    </Card>
  );
}
