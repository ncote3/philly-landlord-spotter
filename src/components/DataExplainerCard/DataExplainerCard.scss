.dataDescCardHeader {
  background-color: #1A1A1A !important;
  color: #fff !important;
  border: 1px solid #fff;
}

.dataDescCardBold {
  font-weight: bold;
}

.dataDescCardItalic {
  font-style: italic;
}