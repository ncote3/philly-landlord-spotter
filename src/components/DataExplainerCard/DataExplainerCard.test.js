import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import DataExplainerCard from "./DataExplainerCard";
import pretty from "pretty";

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("renders correctly with props", () => {
    act(() => {
        render(
            <DataExplainerCard title={'Title'} example={'Example'} description={'Desc'} dataReference={'Data Ref'} />,
            container
        );
    });
    expect(container.innerHTML).toMatchSnapshot(pretty(container.innerHTML));
});