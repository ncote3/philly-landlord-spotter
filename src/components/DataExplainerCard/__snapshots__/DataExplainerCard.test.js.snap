// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props: <div class="card border-dark">
  <div class="dataDescCardBold dataDescCardHeader card-header">Title</div>
  <div class="card-body">
    <p class="card-text">
    <p><span class="dataDescCardBold">Example: </span><span class="dataDescCardItalic">Example</span></p>
    <p><span class="dataDescCardBold">Description:</span>Desc</p>
    <p><span class="dataDescCardBold">Data Reference: </span><span class="dataDescCardItalic">Data Ref</span></p>
    </p>
  </div>
</div> 1`] = `"<div class=\\"card border-dark\\"><div class=\\"dataDescCardBold dataDescCardHeader card-header\\">Title</div><div class=\\"card-body\\"><p class=\\"card-text\\"><p><span class=\\"dataDescCardBold\\">Example: </span><span class=\\"dataDescCardItalic\\">Example</span></p><p><span class=\\"dataDescCardBold\\">Description:</span>Desc</p><p><span class=\\"dataDescCardBold\\">Data Reference: </span><span class=\\"dataDescCardItalic\\">Data Ref</span></p></p></div></div>"`;
